cli/azd/pkg/project/resource_manager.go

```diff
@@ -0,0 +1,220 @@
+package project
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/azureutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/infra"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+)
+
+type ResourceManager interface {
```

Comment:
Another place where I'd love if we just used a struct instead of the interface until we get to a point where mocking this in tests makes sense.

Reply:
I do like exposing the interfaces to support better mocking story for tests.  Now that are components are getting smaller and more loosely coupled creating mocks for these smaller services would allow for tests that do not have to jump down to mocking at the command or HTTP level.